### PR TITLE
chore: added fcmToken to device metadata

### PR DESCRIPTION
**Pull request by ice-hades**

### CODE REVIEW

File: src/store/modules/Devices/sagas/updateDeviceMetadata.ts [hunk 1]
```diff
@@ -91,6 +94,7 @@ export function* updateDeviceMetadataSaga() {
       const deviceUniqueId: ReturnType<typeof deviceUniqueIdSelector> =
         yield select(deviceUniqueIdSelector);
 
+      console.log('pushNotificationToken', pushNotificationToken);
```
Comment (ice-orion) [inline src/store/modules/Devices/sagas/updateDeviceMetadata.ts:97]: console.log

File: src/store/modules/PushNotifications/hooks/useInitNotifications.tsx [hunk 2]
```diff
@@ -23,11 +26,14 @@ export function useInitNotifications() {
         ReactMoE.setupNotificationChannelsAndroid();
       }
     }
+    if (hasPushPermissions) {
+      dispatch(DeviceActions.UPDATE_DEVICE_METADATA.START.create());
```
Comment (ice-orion) [inline src/store/modules/PushNotifications/hooks/useInitNotifications.tsx:30]: Why are you dispatching it from here? As i understand `messaging().getToken()` stays the same no matter if you have or have no push permissions

Comment (ice-hades) [reply src/store/modules/PushNotifications/hooks/useInitNotifications.tsx:30]: @ice-ares btw should we send fcm token if user revoked permissions? Or send an empty string in that case?

Comment (ice-ares) [reply src/store/modules/PushNotifications/hooks/useInitNotifications.tsx:30]: send empty string. Logic is simple:

1. if I can send push notification to user, set the field with the correct value
2. else (if no permissions or whatever reason) set it to empty

Comment (ice-ares) [reply src/store/modules/PushNotifications/hooks/useInitNotifications.tsx:30]: and it should be updated to reflect the latest state. If user revoked permissions, set it to empty string immediately

**Review by ice-orion — DISMISSED** [review]

**Review by ice-hector — APPROVED** [review]

File: src/store/modules/Devices/sagas/updateDeviceMetadata.ts [hunk 3]
```diff
@@ -28,13 +38,18 @@ export function* updateDeviceMetadataSaga() {
...
+    );
+
     const lastUpdateDate: ReturnType<typeof lastMetadataUpdateSelector> =
       yield select(lastMetadataUpdateSelector);
 
     const hoursFromLastUpdate = lastUpdateDate
       ? dayjs().diff(lastUpdateDate, 'hours')
       : 0;
     const shouldUpdateMetadata =
+      action.type === DeviceActions.UPDATE_DEVICE_METADATA.START.type ||
```
Comment (ice-ares) [inline src/store/modules/Devices/sagas/updateDeviceMetadata.ts:52]: when does this happen?

Comment (ice-hades) [reply src/store/modules/Devices/sagas/updateDeviceMetadata.ts:52]: when user grants permission initially or revokes/grants later

File: src/store/modules/Devices/sagas/updateDeviceMetadata.ts [hunk 4]
```diff
@@ -83,6 +99,7 @@ export function* updateDeviceMetadataSaga() {
         DeviceInfo.getInstallerPackageName(),
         DeviceInfo.getBaseOs(),
         DeviceInfo.getBootloader(),
+        hasPushPermissions ? messaging().getToken() : Promise.resolve(''),
```
Comment (ice-orion) [inline src/store/modules/Devices/sagas/updateDeviceMetadata.ts:102]:
```suggestion
        hasPushPermissions ? messaging().getToken() : '',
```

File: src/store/modules/Devices/sagas/updateDeviceMetadata.ts (hunk 3 above)
Comment (ice-orion) [reply src/store/modules/Devices/sagas/updateDeviceMetadata.ts:52]: it's not quite good to rely on an action type that started the saga - just add a param "forceUpdate" and set it to true when you need to bypass the time restrictions and update metadata

**Review by ice-orion — APPROVED** [review]

**Review by ice-hector — DISMISSED** [review]

File: src/store/modules/Devices/actions/index.ts [hunk 5]
```diff
@@ -11,7 +11,7 @@ const UPDATE_SETTINGS = createAction('SET_SETTINGS', {
 });
 
 const UPDATE_DEVICE_METADATA = createAction('SET_DEVICE_METADATA', {
-  START: () => {},
+  START: ({forceUpdate}: {forceUpdate?: boolean}) => ({forceUpdate}),
```
Comment (ice-atlas) [inline src/store/modules/Devices/actions/index.ts:14]: `START: (payload: {forceUpdate?: boolean}) => payload,`

**Review by ice-atlas — CHANGES_REQUESTED** [review]

File: src/store/modules/Devices/actions/index.ts (hunk 5 above)
Comment (ice-hades) [reply src/store/modules/Devices/actions/index.ts:14]: are you serious? Guys need to approve it again because of this?)

Comment (ice-atlas) [reply src/store/modules/Devices/actions/index.ts:14]: imagine to add new param here. 3 places instead of 1

Comment (ice-hades) [reply src/store/modules/Devices/actions/index.ts:14]: 6 secs instead of 2?

Comment (ice-ares) [reply src/store/modules/Devices/actions/index.ts:14]: @ice-hades dont worry about approvals, those will come, if there’s a fix needed, we do it regardless

Comment (ice-orion) [reply src/store/modules/Devices/actions/index.ts:14]: for me it's not a problem to reapprove -> just trigger me if needed

Comment (ice-atlas) [reply src/store/modules/Devices/actions/index.ts:14]: motivation: you can add any param to type and linter will be OK with it, but actual usage will be missed

**Review by ice-atlas — APPROVED** [review]

**Review by ice-orion — APPROVED** [review]

**Pull request merged**